apps/frontend/src/renderer/components/Insights.tsx

```
@@ -415,15 +543,210 @@ function MessageBubble({
+                blockquote: ({ children }: any) => (
+                  <blockquote className="border-l-4 border-primary/50 pl-4 my-3 italic text-muted-foreground">
+                    {children}
+                  </blockquote>
+                ),
+                // קוד
+                code: ({ inline, className, children, ...props }: any) => {
+                  const match = /language-(\w+)/.exec(className || '');
+                  const language = match ? match[1] : '';
+                  const [copied, setCopied] = useState(false);
+                  
+                  const handleCopy = () => {
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)